### PR TITLE
DOC: Improve docs for plotting

### PR DESCRIPTION
**Pull request by bashtage**
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Some doc issues found when adding typing to pandas-stubs

### CODE REVIEW

File: pandas/plotting/_misc.py [hunk 1]
```diff
@@ -285,14 +285,18 @@ def andrews_curves(
...
-    samples : Number of points to plot in each curve
-    color : list or tuple, optional
-        Colors to use for the different classes.
+    class_column : label
+        Name of the column containing class names.
+    ax : axes object, default None
+        Axes to use.
+    samples : int
+        Number of points to plot in each curve.
+    color : {str, list, tuple}, optional
```
Comment (datapythonista) [inline pandas/plotting/_misc.py:294]: My understanding is that we use `{foo, bar}` to indicate that we expect the values `foo` or `bar`. For types, I think we could use `str or Sequence of int, optional`.

File: pandas/plotting/_misc.py [hunk 2]
```diff
@@ -470,19 +474,21 @@ def lag_plot(series: Series, lag: int = 1, ax: Axes | None = None, **kwds) -> Ax
 
     Parameters
     ----------
-    series : Time series
-    lag : lag of the scatter plot, default 1
+    series : Series
+        The time series to visualize.
+    lag : int, optional
```
Comment (datapythonista) [inline pandas/plotting/_misc.py:479]:
```suggestion
    lag : int, default 1
```

We use `optional` when the default values is `None`, meaning not provided (as opposed a `None` value that is used, for example to replace a value by `None`...).

**Review by datapythonista — COMMENTED** [review]
looks good, couple of comments on the conventions I think we've been using.

File: pandas/plotting/_misc.py (hunk 1 above)
Comment (bashtage) [reply pandas/plotting/_misc.py:294]: Fixed, thanks.

File: pandas/plotting/_misc.py (hunk 2 above)
Comment (bashtage) [reply pandas/plotting/_misc.py:479]: Fixed, thanks.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @bashtage

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @bashtage

**Pull request merged**